include/zephyr/net/lwm2m.h

```diff
@@ -1261,6 +1261,11 @@ enum lwm2m_rd_client_event {
 	LWM2M_RD_CLIENT_EVENT_NETWORK_ERROR,
 };
 
+/**
+ *	Define for old event name keeping backward compability.
+ */
+#define LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE LWM2M_RD_CLIENT_EVENT_REG_TIMEOUT
```

Comment:
Could use `__DEPRECATED_MACRO` here so that we can remove the old name eventually.

Reply:
That's a good idea.